File: backend/src/main/java/com/foodly/foodlyapi/enums/OrderStatus.java
package com.foodly.foodlyapi.enums;

import lombok.Getter;

@Getter
public enum OrderStatus {
    PENDING(0, "Beklemede"),
    PAYMENT_APPROVED(1, "Ödeme Onaylandı"),
    ORDER_APPROVED(2, "Sipariş Onaylandı"),
    PREPARING(3, "Hazırlanıyor"),
    DELIVERED(4, "Teslim Edildi"),
    REJECTED(5, "Reddedildi");

    private final int index;
    private final String description;

    OrderStatus(int index, String description) {
        this.index = index;
        this.description = description;
    }
}

File: frontend/src/constants/OrderStatus.js
export const OrderStatus = {
  PENDING: {
    value: 'PENDING',
    description: 'Beklemede'
  },
  PAYMENT_APPROVED: {
    value: 'PAYMENT_APPROVED',
    description: 'Ödeme Onaylandı'
  },
  ORDER_APPROVED: {
    value: 'ORDER_APPROVED',
    description: 'Sipariş Onaylandı'
  },
  PREPARING: {
    value: 'PREPARING',
    description: 'Hazırlanıyor'
  },
  READY_FOR_DELIVERY: {
    value: 'READY_FOR_DELIVERY',
    description: 'Teslimata Hazır'
  },
  DELIVERED: {
    value: 'DELIVERED',
    description: 'Teslim Edildi'
  },
  REJECTED: {
    value: 'REJECTED',
    description: 'Reddedildi'
  }
};

// Helper function to get description
export const getOrderStatusDescription = (status) => {
  return OrderStatus[status]?.description || status;
};
